### PR TITLE
[Fix] Fix Facebook signup crash

### PR DESCRIPTION
**Pull request by mennenia**
This is an applied fix to 3.0.2-2 for #2056.

In the process of removing the `MoreInfoViewController`, part of the login code for Facebook got deleted also, which meant that upon return from the Facebook app / requesting a FB token, we then wouldn't log the user in. This meant that `ARUserManager` did not have a current user, which caused the crash in `ARAppDelegate+Emission.m`.

Personally, I've tested this fix, on device, in the following scenarios:
- signup, new user, facebook app  
- signup, new user, safari modal login
- signup, existing user, facebook app (onboarding follows get applied)
- signup, existing user, safari modal login (onboarding follows get applied)
- login, existing user, facebook app
- login, existing user, safari modal login
- login, existing user who removed Artsy from FB 'Apps', facebook login - can still log in
- login, existing user who removed Artsy from FB 'Apps', safari modal login - can still log in
- login, complete new user, safari modal login - errors, rightly so

### CODE REVIEW

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m [hunk 1]
```diff
@@ -73,7 +80,10 @@ - (void)viewWillAppear:(BOOL)animated
 - (void)viewDidAppear:(BOOL)animated
 {
     [super viewDidAppear:animated];
-    [self.textFieldsView.nameField becomeFirstResponder];
+
+    if (!_movedToFacebook) {
```
Comment (alloy) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: Prefer property getter usage over direct ivar access.

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m [hunk 2]
```diff
@@ -358,7 +369,102 @@ - (void)viewWillDisappear:(BOOL)animated
...
+
+
+- (void)fbError
+{
+    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Couldn’t get Facebook credentials"
+                                                    message:@"Couldn’t get Facebook credentials. Please link a Facebook account in the settings app. If you continue having trouble, please email Artsy support at support@artsy.net"
+                                                   delegate:self
+                                          cancelButtonTitle:@"OK"
+                                          otherButtonTitles:nil];
+    [alert show];
```
Comment (alloy) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: Btw, a pet peeve of mine is that you also get to see this banner if you decided to cancel from FB’s interface.

File: CHANGELOG.yml [hunk 3]
```diff
@@ -1,9 +1,26 @@
 upcoming:
-    version: 3.0.2
-    details: Second patch fix for 3.0.0 release.
+    version: 3.1.0
```
Comment (mennenia) [inline CHANGELOG.yml]: I've based the changelog off of master, because otherwise it would end in a merge conflict

File: CHANGELOG.yml [hunk 4]
```diff
@@ -1,9 +1,26 @@
...
     dev:
       infrastructure:
+        - Updates to Swift 3 / Xcode 8 - ash/orta
+        - Removes the Apple Watch app - orta
+        - Remove extra xapp fetching and old home feed - maxim
+    user_facing:
+      - Fixes issue where links in auction descriptions would lead nowhere when tapped - ash
+
+releases:
+  - version: 3.0.2
```
Comment (mennenia) [inline CHANGELOG.yml]: Is this still 3.0.2? or does something like 3.0.2.1 exist?

**Comment by mennenia** [comment]
@alloy Done, and will address the popup copy in 3.1.0

File: CHANGELOG.yml (hunk 3 above)
Comment (alloy) [inline CHANGELOG.yml]: Ah but this unfortunately can’t be done in this way. Your change needs to go in a upcoming 3.0.3 section, that version is then used by the beta build process to determine the version number.

File: CHANGELOG.yml (hunk 4 above)
Comment (alloy) [inline CHANGELOG.yml]: 3.0.3

**Comment by alloy** [comment]
Looks good 👍 

So yeah, we can’t merge into master easily, but we can worry about that after the release.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">You need to set the App’s plist version to 3.1.0</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |244.7ms | [AuctionViewController.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Auction/AuctionViewController.swift#L199) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
192.8ms | [SwiftyJSON.swift](/artsy/eigen/tree/mc-facebook-fix/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
173.2ms | [WebSocket.swift](/artsy/eigen/tree/mc-facebook-fix/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
89.0ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L47) | @objc override func viewDidLoad()
82.9ms | [Observable.swift](/artsy/eigen/tree/mc-facebook-fix/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
73.2ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
71.2ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
64.3ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
62.2ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/mc-facebook-fix/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by alloy** [comment]
> You need to set the App’s plist version to 3.1.0

What’s that about?

**Comment by alloy** [comment]
Danger was probably complaining about the CHANGELOG not matching the Info.plist https://github.com/artsy/eigen/pull/2058/commits/6a2f9eba01ef51ceb02b17b395b3f7253a78c901

**Comment by mennenia** [comment]
Ohhh sorry, I didn't know about changing the plist too (although that makes total sense).

Will keep in mind!

**Comment by mennenia** [comment]
@orta could you have a look at this? This was the hotfix we applied and released just before the holidays, and if we're pushing out a build for auctions soon, this will need to be in master. I've updated it, so there should be no merge conflicts.

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m [hunk 5]
```diff
@@ -358,7 +369,102 @@ - (void)viewWillDisappear:(BOOL)animated
...
+                                                              }
+                                                              
+                                                              // something else went wrong
+                                                              ARErrorLog(@"Couldn't link Facebook account. Error: %@. The server said: %@", error.localizedDescription, JSON);
+                                                              
+                                                              NSString *errorMessage = [NSString stringWithFormat:@"Server replied saying '%@'.", JSON[@"error"] ?: JSON[@"message"] ?: error.localizedDescription];
+                                                              
+                                                              // we'll display an alert view
+                                                              [sself showErrorAlertWithMessage:errorMessage];
+        }];
```
Comment (orta) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: argh, this style of indentation is the worst ( I know it's spacecommander but realistically the fault lies at Apple and the obj-c style they encouraged )

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m [hunk 6]
```diff
@@ -358,7 +369,102 @@ - (void)viewWillDisappear:(BOOL)animated
...
+                                        // TODO: handle this
+                                        __strong typeof (wself) sself = wself;
+                                        [sself ar_removeIndeterminateLoadingIndicatorAnimated:YES];
+        }
+        networkFailure:^(NSError *error) {
+                                               // TODO: handle this
+                                               __strong typeof (wself) sself = wself;
+                                               [sself ar_removeIndeterminateLoadingIndicatorAnimated:YES];
+        }];
+}
```
Comment (orta) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: I mean look at this crazy indentation ^ 🍡

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m (hunk 2 above)
Comment (orta) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: There is code protecting for that use case, I wonder if their errors have changed 

`+        if (![reason isEqualToString:@"com.facebook.sdk:UserLoginCancelled"]) {`

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m (hunk 6 above)
Comment (mennenia) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/3_-_Sign_Up_+_Log_In/ARCreateAccountViewController.m]: I know, I tried to separate cases and bits, but alas :(.

**Pull request merged**